### PR TITLE
[test] separate code coverage data files

### PR DESCRIPTION
**Pull request by bukepo**
This commit ensures each node create its own profile data, so that they
can be merged after tests to produce more accurate profile data.

This commit also prints the coverage data of message.cpp after all
profile data are merged.

### CODE REVIEW

File: tests/scripts/expect/_common.exp [hunk 1]
```diff
@@ -69,25 +70,31 @@ proc spawn_node {id {type ""} {radio_url ""}} {
 
     send_user "\n# ${id} ${type}\n"
 
+    if {[info exists ::env(CC)] && $::env(CC) == "clang"} {
```
Comment (Moandor-y) [inline tests/scripts/expect/_common.exp:73]: It seems `llvm-cov` supports `GCOV_PREFIX` too. <https://llvm.org/docs/CommandGuide/llvm-cov.html>

Any chance to make this also applied to LLVM?

**Review by Moandor-y — APPROVED** [review]
LGTM overall :+1: 

With a question/suggestion below

File: tests/scripts/expect/_common.exp (hunk 1 above)
Comment (bukepo) [reply tests/scripts/expect/_common.exp:73]: But there's no `gcov-tool merge` for `llvm-cov`

Comment (Moandor-y) [reply tests/scripts/expect/_common.exp:73]: OK, I see. Look good.

**Review by jwhui — APPROVED** [review]

**Pull request merged**